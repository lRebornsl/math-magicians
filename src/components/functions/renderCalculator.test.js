import React from 'react';
import { render } from '@testing-library/react';
import RenderCalculator from './renderCalculator';

describe('RenderCalculator', () => {
  it('renders correctly', () => {
    const { container } = render(<RenderCalculator />);
    expect(container).toMatchSnapshot();
  });

  it('renders all buttons', () => {
    const { container } = render(<RenderCalculator />);
    const buttons = container.querySelectorAll('button');
    expect(buttons.length).toBe(19);
  });
});
