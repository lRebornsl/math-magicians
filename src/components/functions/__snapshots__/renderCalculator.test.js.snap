// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCalculator renders correctly 1`] = `
<div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      AC
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      +/-
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      %
    </button>
  </div>
  <div
    class="flex calcItem  itemOperation"
  >
    <button
      type="button"
    >
      รท
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      7
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      8
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      9
    </button>
  </div>
  <div
    class="flex calcItem  itemOperation"
  >
    <button
      type="button"
    >
      x
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      4
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      5
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      6
    </button>
  </div>
  <div
    class="flex calcItem  itemOperation"
  >
    <button
      type="button"
    >
      -
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      1
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      2
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      3
    </button>
  </div>
  <div
    class="flex calcItem  itemOperation"
  >
    <button
      type="button"
    >
      +
    </button>
  </div>
  <div
    class="flex calcItem itemZero "
  >
    <button
      type="button"
    >
      0
    </button>
  </div>
  <div
    class="flex calcItem  "
  >
    <button
      type="button"
    >
      .
    </button>
  </div>
  <div
    class="flex calcItem  itemOperation"
  >
    <button
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
